cpp/csp/python/PyEngine.h

```diff
@@ -54,6 +55,21 @@ class CSPIMPL_EXPORT PyEngine final: public PyObject
     Engine    * m_engine;
 };
 
+// Generic engine shutdown function for push-type adapters
+template<typename T>
+PyObject * PyEngine_shutdown( T * self, PyObject * args, PyObject * kwargs )
+{
+    CSP_BEGIN_METHOD;
+
+    const char * msg;
```

Comment:
so if we do try to pass the exception info through, we can decide to either take a string or the exception info here ( in case there is a case where a user wants to shutdown for some other reason and doesnt have an exception handy )

Reply:
In the Python wrapper (public method) we can also just create an `Exception('msg')` if a string is passed, and then the C++ impl only ever sees the Exception.

Reply:
yes that would be the only way to do it

Reply:
They can always just create an `Exception('str')` on their end. 
Might be less confusing than allowing both string and exceptions to be passed.